### PR TITLE
replace validate_* methods with datatypes

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: manifests/gemrc.pp [hunk 1]
```diff
@@ -33,32 +33,19 @@
...
-  $gemrc                      = $::ruby::params::gemrc,
-  $owner                      = 'root',
-  $group                      = 'root',
-  $mode                       = '0644',
-  $disable_default_gem_server = undef
+  $sources                                              = undef,
+  Optional[Variant[Boolean, Enum[':really']]] $verbose  = undef,
+  Optional[Boolean] $update_sources                     = undef,
+  Optional[Boolean] $backtrace                          = undef,
+  $gempath                                              = undef,
```
Comment (igalic) [inline manifests/gemrc.pp]: why don't we use Stdlib::Absolutepath here?

File: manifests/gemrc.pp [hunk 2]
```diff
@@ -33,32 +33,19 @@
...
+  $gem_command                                          = undef,
+  $gemrc                                                = $::ruby::params::gemrc,
+  $owner                                                = 'root',
+  $group                                                = 'root',
+  $mode                                                 = '0644',
```
Comment (igalic) [inline manifests/gemrc.pp]: `Pattern[[0-7]{4}]`

File: manifests/gemrc.pp [hunk 3]
```diff
@@ -33,32 +33,19 @@
...
-  $disable_default_gem_server = undef
+  $sources                                              = undef,
+  Optional[Variant[Boolean, Enum[':really']]] $verbose  = undef,
+  Optional[Boolean] $update_sources                     = undef,
+  Optional[Boolean] $backtrace                          = undef,
+  $gempath                                              = undef,
+  $gem_command                                          = undef,
+  $gemrc                                                = $::ruby::params::gemrc,
+  $owner                                                = 'root',
+  $group                                                = 'root',
```
Comment (igalic) [inline manifests/gemrc.pp]: These should all just be `String`, except for `gem_command`, which should be `Optional[String]`

File: manifests/gemrc.pp (hunk 2 above)
Comment (bastelfreak) [inline manifests/gemrc.pp]: fixed

**Comment by tux-o-matic** [comment]
@bastelfreak using Enum on Ruby dev package ensure is too restrictive. Some use cases requires ensure to be set to a specific version.

**Pull request merged**